spec/components/blacklight/gallery/slideshow_component_spec.rb

```diff
@@ -15,20 +15,29 @@
     Capybara::Node::Simple.new(render)
   end
 
-  let(:blacklight_config) { Blacklight::Configuration.new }
+  let(:document) do
+    SolrDocument.new(
+      id: 'x',
+    )
+  end
+
   let(:presenter) { Blacklight::IndexPresenter.new(document, view_context, blacklight_config) }
 
   before do
+    allow(view_context).to receive(:current_search_session).and_return(nil)
+    allow(view_context).to receive(:search_session).and_return({})
     allow(view_context).to receive(:blacklight_config).and_return(blacklight_config)
   end
 
   describe '#slideshow_tag' do
     subject { rendered }
 
-    let(:document) { SolrDocument.new({}) }
-
     context 'with a slideshow method' do
-      let(:blacklight_config) { Blacklight::Configuration.new.tap { |config| config.index.slideshow_method = :xyz } }
+      let(:blacklight_config) { Blacklight::Configuration.new.tap { 
```

Comment:
The old one-liners were hard to read, so I like the idea of making them multi-line. I'd suggest using `do/end` for clarity (and keeping the block argument with the beginning of the block), e.g.

```
let(:blacklight_config) do
  Blacklight::Configuration.new.tap do |config|
    # ...
  end
end
```